### PR TITLE
pinmux: sam0: add a pinmux driver for the SAM0 series

### PR DESCRIPTION
**Pull request by nzmichaelh**
Pin multiplexing is a function of the PORT peripheral.  This change defines a separate pinmux device at the same address as the PORTs themselves.

@sknsean FYI

### CODE REVIEW

File: drivers/pinmux/pinmux_sam0.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+struct pinmux_sam0_config {
+	PortGroup *regs;
+};
+
+static int pinmux_sam0_set(struct device *dev, u32_t pin, u32_t func)
+{
+	const struct pinmux_sam0_config *cfg = dev->config->config_info;
+
+	if ((pin & 1) == 0) {
```
Comment (galak) [inline drivers/pinmux/pinmux_sam0.c]: Can we get rid of the magic numbers here?

File: drivers/pinmux/pinmux_sam0.c [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	cfg->regs->PINCFG[pin].bit.PMUXEN = 1;
+
+	return 0;
+}
+
+static int pinmux_sam0_get(struct device *dev, u32_t pin, u32_t *func)
+{
+	const struct pinmux_sam0_config *cfg = dev->config->config_info;
+
+	if ((pin & 1) == 0) {
```
Comment (galak) [inline drivers/pinmux/pinmux_sam0.c]: same magic # comment.

**Review by galak — CHANGES_REQUESTED** [review]

File: drivers/pinmux/pinmux_sam0.c (hunk 1 above)
Comment (nzmichaelh) [reply drivers/pinmux/pinmux_sam0.c]: Done.  Added a comment explaining the packing and pulled the args out to variables.

File: drivers/pinmux/pinmux_sam0.c (hunk 2 above)
Comment (nzmichaelh) [reply drivers/pinmux/pinmux_sam0.c]: Done

**Review by galak — APPROVED** [review]

**Comment by sknsean** [comment]
Great job :-)

**Pull request merged**